OloEngine/tests/Lua/LuaBindingTest.cpp

```
@@ -0,0 +1,1295 @@
+    auto result = lua.script("local v = vec4.new(1.0, 2.0, 3.0, 4.0); return v.x, v.y, v.z, v.w");
+    EXPECT_FLOAT_EQ(result.get<f32>(0), 1.0f);
+    EXPECT_FLOAT_EQ(result.get<f32>(1), 2.0f);
+    EXPECT_FLOAT_EQ(result.get<f32>(2), 3.0f);
+    EXPECT_FLOAT_EQ(result.get<f32>(3), 4.0f);
+}
+
+// =============================================================================
+// TransformComponent
+// =============================================================================
+
+TEST_F(LuaBindingTest, TransformComponent_TranslationRoundTrip)
```

Clang (14.0.6)

[warning] use a trailing return type for this function

(modernize-use-trailing-return-type)